pf9/support/generate_bundle.py

```diff
@@ -0,0 +1,201 @@
+                    self.upload_pf9cli_logs(du_url, host)
+
+        else:
+            # Generating support Bundle for remote Host
+            ssh_dir = os.path.join(os.path.expanduser("~"), '.ssh/id_rsa')
+            user_name = getpass.getuser()
+            ssh_conn = Connection(host=host,
+                                      user=user_name,
+                                      port=22)
+            try:
+                ssh_result_generate = ssh_conn.sudo(bundle_exec, hide='stderr')
+                ssh_result_bundle = ssh_conn.sudo('ls -sh /tmp/pf9-support.tgz > /dev/null 2>&1', hide='stderr')
+                if ssh_result_generate.exited and ssh_result_bundle.exited:
+                    ssh_conn.close()
+                time.sleep(2)
+                self.remote_host_upload(du_url, ssh_conn, host)
+                self.upload_pf9cli_logs(du_url, host)
+
+            except paramiko.ssh_exception.NoValidConnectionsError:
+                    self.upload_pf9cli_logs(du_url, host)
+                    sys.exit(1)
+                #If passwordless ssh is not configured to a host , use the username and password provided in the prep node command to generate the log bundle
+            except (paramiko.ssh_exception.SSHException,paramiko.ssh_exception.PasswordRequiredException,invoke.exceptions.AuthFailure) as err:
+                    user_name = user
+                    password = password
+                    ssh_auth = {"look_for_keys": "false", "password": password}
+                    config = {'password': password}
+                    try:
+                        ssh_conn = Connection(host=host,
+                                              user=user_name,
+                                              port=22,
+                                              connect_kwargs=config)
+                        ssh_result_generate = ssh_conn.sudo(bundle_exec, hide='stderr')
+                        ssh_result_bundle = ssh_conn.sudo('ls -sh /tmp/pf9-support.tgz > /dev/null 2>&1', hide='stderr')
+                        if ssh_result_generate.exited and ssh_result_bundle.exited:
+                           ssh_conn.close()
+                        time.sleep(2)
+                        self.remote_host_upload(du_url, ssh_conn, host)
+                        self.upload_pf9cli_logs(du_url, host)
+                    except (paramiko.ssh_exception.SSHException,paramiko.ssh_exception.PasswordRequiredException,invoke.exceptions.AuthFailure) as err:
+                        self.upload_pf9cli_logs(du_url, host)
+
+        return None
+
+
+    def upload_logs(self, du_url):
+        #function to upload the logs from the localhost to S3
+        host = socket.getfqdn()
+        du_url = du_url.replace("https://","")
+        filename = "/tmp/pf9-support.tgz"
+        header = 'x-amz-acl:bucket-owner-full-control'
+        S3_location = "https://s3-us-west-2.amazonaws.com/loguploads.platform9.com/"+str(du_url)+"/"+str(host)+"/"
```

Comment:
You can define the hard coded locations and paths at the top of the file. For example,
```
S3_LOCATION = "https://s3-us-west-2.amazonaws.com/loguploads.platform9.com/{}/{}/"
```
and use it here as
```
s3_location = S3_LOCATION.format(du_url, host)
```
For other strings, just define them at the top of the file.